### PR TITLE
new: Added a new lab 'Mountpoint for S3'

### LINKED ISSUE #843: Add a Mountpoint for Amazon S3 Module to the Fundamentals/Storage exercises
### Outline the high level topic of the lab

This module will walk the user through the process of installing the CSI driver for Mountpoint for Amazon S3 and then using the CSI driver to provision storage on S3 using Mountpoint for file level access in EKS. 

### Provide the flow of the lab exercise, including what sample application components will be used

-     IAM roles for CSI driver created in Terraform
-     S3 bucket created in Terraform
-     CSI Driver installed manually from the repo of the CSI driver
-     Storage Class deployed manually 
-     PVC created manually 
-     Deploy an nginx pod using a deployment and add a new product sku in S3 via Mountpoint
-     Clean up the environment using Terraform

### What additional AWS infrastructure or EKS addons/components will need to be created or installed to support this lab?

- S3 bucket
- IAM role for CSI driver

### What additional software or configuration will be required in the Cloud9 IDE?

None

### Are enhancements to the retail sample application required to support this lab exercise?

TBD

### PR DESCRIPTION
**Pull request by rchandra20**
#### What this PR does / why we need it:

EKS Workshop can benefit greatly from a module that showcases the use of Mountpoint for S3. This content is placed under the Fundamentals/Storage exercises. 

#### Quality checks

- [x] My content adheres to the style guidelines
- [x] I ran `make test module="fundamentals/storage/mountpoint-s3"` it was successful (see https://github.com/aws-samples/eks-workshop-v2/blob/main/docs/automated_tests.md)
- [x] The PR has meaningful title and description of the changes that will be included in the workshop release notes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by heinrichse — DISMISSED** [review]
Reviewed and approved PR

**Review by heinrichse — DISMISSED** [review]

**Review by heinrichse — DISMISSED** [review]
Reviewed and approved

**Review by pengc99 — COMMENTED** [review]
Reviewed this in rchandra20's repo already - this content is ready to be merged from my end. Would appreciate a second look.

**Review by niallthomson — CHANGES_REQUESTED** [review]
Looks great overall, noted some changes I think we should make

File: website/docs/fundamentals/storage/mountpoint-s3/30-persistent-object-storage-with-mountpoint-for-amazon-s3.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+sidebar_position: 30
+---
+
+Remember that our end goal is to have an image host application that **scales horizontally** and has **persistent storage** backed by Amazon S3. In the previous steps we created a staging directory for our image objects, then we downloaded the image assets into the staging directory and uploaded them into our S3 bucket. Finally, we installed the Mountpoint for Amazon S3 CSI driver and added it to our environment. We now need to attach our pods to use this PV provided by the Mountpoint for Amazon S3 CSI driver.
+
+Let's create a [Persistent Volume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) and change the `assets` container on the assets deployment to mount the Volume created.
+
+First inspect the `s3pvclaim.yaml` file to see the parameters in the file and the claim:
+
+```file
```
[the remark this answers is not in the document]
Comment (rchandra20) [reply website/docs/fundamentals/storage/mountpoint-s3/30-persistent-object-storage-with-mountpoint-for-amazon-s3.md:12]: [Removed comments on s3pvclaim.yaml](https://github.com/aws-samples/eks-workshop-v2/pull/1176/commits/b9e0f10731d32d9f3df210e43fa0ecb3d85f1643)

File: website/docs/fundamentals/storage/mountpoint-s3/10-ephemeral-container-storage.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+sidebar_position: 10
+---
+
+We have a deployment already borrowed from the simple store example but we will modify it slightly to repurpose this scenario as an image host. The assets microservice utilizes a webserver running on EKS. Web servers are a great example for the use of deployments because they **scale horizontally** and **declare the new state** of the Pods.
+
+The assets component is a container which serves static images for products, these product images are added as part of the container image build. However with this setup, images submitted on one container do not propagate to another container. In this exercise we'll utilize [Amazon S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/Welcome.html) and Kubernetes [Persistent Volume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) to setup a common storage environment where assets like images can get served with multiple web server containers that scale with demand.
+
+We can start by describing the Deployment to take a look at its initial volume configuration:
+
+```bash
```
[the remark this answers is not in the document]
Comment (rchandra20) [reply website/docs/fundamentals/storage/mountpoint-s3/10-ephemeral-container-storage.md:12]: Trimmed down deployment section in 'Ephemeral Container Storage'

File: website/docs/fundamentals/storage/mountpoint-s3/30-persistent-object-storage-with-mountpoint-for-amazon-s3.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+AME       STATUS   VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS   VOLUMEATTRIBUTESCLASS   AGE
+s3-claim   Bound    s3-pv    1Gi        RWX                           <unset>                 6m41s
+
+$ kubectl get pods -n assets
+NAME                     READY   STATUS    RESTARTS   AGE
+assets-9fbbbcd6f-26jrq   1/1     Running   0          6m57s
+assets-9fbbbcd6f-lb46c   1/1     Running   0          6m55s
+```
+
+Finally, let's take a look at our final deployment with the Mountpoint for Amazon S3 CSI driver:
```
[the remark this answers is not in the document]
Comment (rchandra20) [reply website/docs/fundamentals/storage/mountpoint-s3/30-persistent-object-storage-with-mountpoint-for-amazon-s3.md:95]: [Used YAML UI component, described PVC, and trimmed deployment informa…](https://github.com/aws-samples/eks-workshop-v2/pull/1176/commits/f924839b85389fa9f72253445a623d9ddf909d96) 
[f924839](https://github.com/aws-samples/eks-workshop-v2/pull/1176/commits/f924839b85389fa9f72253445a623d9ddf909d96)
…tion

**Pull request merged**